### PR TITLE
kubeadm: add e2e test for UpgradeAddonsBeforeControlPlane feature gate

### PR DESCRIPTION
**Pull request by SataQiu**
kubeadm: add e2e test for UpgradeAddonsBeforeControlPlane feature gate

Ref: kubernetes/kubeadm#2346

### CODE REVIEW

File: config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kinder.yaml [hunk 1]
```diff
@@ -46,7 +46,7 @@ periodics:
           cpu: 2000m
 
 - name: ci-kubernetes-e2e-kubeadm-kinder-1-27
-  cluster: eks-prow-build-cluster
+  cluster: k8s-infra-prow-build
   interval: 12h
```
Comment (neolit123) [inline config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kinder.yaml:50]: this is expected.
sig k8s infra folks wanted to have one kubeadm job on the eks cluster, but our tools override it.

perhaps one day we can switch to eks for all jobs

**Review by neolit123 — COMMENTED** [review]
/lgtm

File: config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kinder.yaml (hunk 1 above)
Comment (SataQiu) [reply config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kinder.yaml]: Reverted.

**Review by neolit123 — COMMENTED** [review]
/lgtm

File: config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kinder-upgrade-addons-before-controlplane.yaml [hunk 2]
```diff
@@ -0,0 +1,46 @@
+# AUTOGENERATED by https://git.k8s.io/kubeadm/kinder/ci/tools/update-workflows
+periodics:
+
+- name: ci-kubernetes-e2e-kubeadm-kinder-upgrade-addons-before-controlplane-1-27-latest
+  cluster: k8s-infra-prow-build
+  interval: 2h
```
Comment (pacoxu) [inline config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kinder-upgrade-addons-before-controlplane.yaml:6]: Not sure if the interval is too short.
24h or 12h would be enough for this feature?

Comment (neolit123) [reply config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kinder-upgrade-addons-before-controlplane.yaml:6]: it's common for our "latest" test jobs

Comment (pacoxu) [reply config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kinder-upgrade-addons-before-controlplane.yaml:6]: Then that is fine.
- The current CI triggering is not that smart to do some back-off strategy.

Comment (neolit123) [reply config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kinder-upgrade-addons-before-controlplane.yaml:6]: yes, we can spread out latest jobs less often in the generator e.g. 4 hours at some future release

**Pull request merged**